player/common/src/main/kotlin/com/tidal/sdk/player/common/model/MediaProduct.kt

```diff
@@ -17,5 +17,6 @@ data class MediaProduct(
     override val productId: String,
     override val sourceType: String? = null,
     override val sourceId: String? = null,
+    override val extras: Map<String, String?> = emptyMap(),
```

Comment:
Similar minor clarification.

Any reason why you went with a default empty Map instead of `null`?

On iOS I'm making this a `let`, meaning a non-modifiable property, so I don't care about easier addition of values to the Map later.

Reply:
Can't remember why I wouldn't have made it nullable tbh, and at a first glance it looks like it would be better. I will change it and see if it shows a reason why it isn't.

Regarding mutability, I don't think it's a good idea to allow additions after having made the call. It will make it impossible to know whether a value added by the host after creating the MediaProduct is included in an event, and also open the possibility for Player to modify the map, which should never occur.

Reply:
I'm agreeing on the mutability part, but I might have explained it badly.

I meant to say that the only benefit I see from always having an `emptyMap()` is if you wanted to add keys later (no need to check for `null`) but because we don't want't to touch the extras map in any way, just forward it, we don't need that "benefit".

Reply:
Funny enough, the map created by `emptyMap()` is immutable, so knowing this and with your clarification, there's even less of a reason for the param to not be nullable 😅

Reply:
I've now finished the conversion to making the parameter nullable and it seems to work fine, let's go with that?

Reply:
I can't remember to be honest. @ziad-halabi9 can you confirm?

Reply:
Yes keep `sourceType` and `sourceId` outside `extras`. They have to live there for a while too.

Reply:
Do we want to add extras to the spec officially, and to the documentation? (considering it is only for internal use at TIDAL)
I don't mind either way, but I ask so we don't forget.

Reply:
I will add it [here](https://github.com/tidal-music/tidal-sdk/blob/main/Player.md) once the work is done on the clients.

Reply:
Yes